### PR TITLE
Add a Python script example using the Python Api client

### PR DESCRIPTION
**Pull request by aine-rb**
This script aims to be the most generic and explicit possible, in order to show how to use the Python API client for automating ZAP.

### CODE REVIEW

**Comment by jangelesg** [comment]
#!/usr/bin/env python
# coding=utf-8
# Author: Jonathan Angeles
# Objective: ZAP provides a REST Application Programming Interface (API) that zap_client
# use to interact with ZAP programmatically.
# Dependencies request 'The ZAP API'
# Usage: python zap_client.py
# Client API Zap Proxy

import sys
import time
from pprint import pprint
from zapv2 import ZAPv2
import urllib
import urllib2
import requests


# Defining functions
def proxy_setup():
    proxies = {}
    zap_proxy_ip = raw_input('Enter the ip proxy: ')
    zap_proxy_port = raw_input('Enter the zap port: ')
    zap_addr = 'http://' + zap_proxy_ip + ':' + zap_proxy_port
    proxies['http'] = zap_addr
    proxies['https'] = zap_addr
    return proxies
#def api_lib_inst ():
    # Download the proper library

    # install the library

# defining the ZAP client

print'********************************************************************'
print'*  Application Security Team                                       *'
print'* Zap_Client for CI Security Application Assessments               *'
print'* By default ZAP API client will connect to port 8080 on 127.0.0.1 *'
print'********************************************************************\n'
try:
    zap = ZAPv2()
    assert isinstance(zap.core.version, object)
    check_zap = zap.core.version

except:
    print 'It seems ZAP Proxy is not configure by default\n'
    proxy_change = raw_input('Would you like to configure the proxy settings? y/n ')
    str(proxy_change)
    if proxy_change == 'y' or proxy_change == 'Y':
        zap = ZAPv2(proxies=proxy_setup())
        print 'Using ZAP Proxy Version %s ' % zap.core.version
    else:
        pass

target = raw_input('Enter the URL to attack: ')
if len(target) < 1:
    target = 'https://www.target-url.com/apply/checking?productid=0027'
# do stuff
print 'Accessing target %s' % target
# try have a unique enough session...
zap.urlopen(target)
# Give the sites tree a chance to get updated
time.sleep(2)

print 'Spidering target %s' % target
scanid = zap.spider.scan(target)
# Give the Spider a chance to start
time.sleep(2)
while (int(zap.spider.status(scanid)) < 100):
    print 'Spider progress %: ' + zap.spider.status(scanid)
    time.sleep(2)

print 'Spider completed'
# Give the passive scanner a chance to finish
time.sleep(5)

print 'Scanning target %s' % target
scanid = zap.ascan.scan(target)
while (int(zap.ascan.status(scanid)) < 100):
    print 'Scan progress %: ' + zap.ascan.status(scanid)
    time.sleep(5)

print 'Scan completed'

# Report the results

print 'Hosts: ' + ', '.join(zap.core.hosts)
print 'Alerts: '
time.sleep(2)
pprint(zap.core.alerts())

**Comment by aine-rb** [comment]
@jangelesg What does this comment mean ?

**Comment by jangelesg** [comment]
Do not pay attention to that, it does not mean nothing .. 

On 2016-08-20 11:26, aine-rb wrote:

> @jangelesg [1] What does this comment mean ? 
> 
> --
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub [2], or mute the thread [3].
 

Links:
------
[1] https://github.com/jangelesg
[2]
https://github.com/zaproxy/zap-api-python/pull/6#issuecomment-241209411
[3]

**Comment by aine-rb** [comment]
Script updated.
Spider, Ajax Spider and Active scans are arranged in a better way.
User can upload a proxy script.
User can choose between a whitelist scan policy or a blacklist one.

**Comment by jangelesg** [comment]
Got cha!!!!

File: src/examples/zap_example_api_script.py [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+authMethod = 'scriptBasedAuthentication'
+
+# MANDATORY only if authMethod is set to scriptBasedAuthentication.
+# Ignored otherwise
+authScriptName = 'NashornTwoStepAuthentication.js'
+# Script engine values: Oracle Nashorn in Java 8 for Javascript, Rhino
+# in Java <= 7 for Javascript, jython for python, JSR 223 JRuby Engine for ruby
+authScriptEngine = 'Oracle Nashorn'
+# Absolute local path
+authScriptFileName = '/home/aine/scripts/authentication/NashornTwoStepAuthentication.js'
```
Comment (thc202) [inline src/examples/zap_example_api_script.py]: This could use a generic path (as above with the other script).

**Comment by thc202** [comment]
@aine-rb would you mind enable "Allow edits from maintainers" so that we can address the remaining comment and merge?

**Comment by aine-rb** [comment]
@thc202 Done. Sorry for this remaining comment, I thought that I had already fixed it.

**Comment by thc202** [comment]
Thanks, no problem!

Rebased and addressed the remaining comment.

**Pull request merged**